### DISCUSSION TITLE
Incorrect type for `addVariant()`

### DISCUSSION THREAD

**Discussion by programming-with-ia** [tailwindlabs/tailwindcss#15405] [Help]
when working with typescript face correct typing issue

when using like this
```js
addVariant(`selector`, (({ modifySelectors, separator, }) => {
    modifySelectors(({ className }) => {
        return `.${e(`selector${separator}${className}`)}[type="${type}"]`;
    });
}));
```

**Comment by philipp-spiess** [comment]
Hey! That's because the APIs you're trying to use are outdated and should not for a while already. Check out the Variant API added in Tailwind CSS v3 to understand how to extend variants: https://tailwindcss.com/docs/plugins#adding-variants

**Reply by programming-with-ia** [reply]
Please add correct `types` for `addVariant()`
